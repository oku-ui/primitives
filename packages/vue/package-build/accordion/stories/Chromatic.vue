<script setup lang="ts">
import { ref } from 'vue'
import { OkuAccordion, OkuAccordionContent, OkuAccordionHeader, OkuAccordionItem, OkuAccordionTrigger } from '@oku-ui/accordion'

const values = ref(['One', 'Two', 'Three', 'Four'])
</script>

<template>
  <div>
    <h1>Uncontrolled</h1>
    <h2>Single closed</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" default-value="Two">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple closed</h2>
    <OkuAccordion type="multiple" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :default-value="['One', 'Two']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Controlled</h1>
    <h2>Single open</h2>
    <OkuAccordion type="single" class="accordion-root" value="Three">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Multiple open</h2>
    <OkuAccordion type="multiple" class="accordion-root" :value="['Two', 'Three']">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (whole)</h1>
    <OkuAccordion type="single" class="accordion-root" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Disabled (item)</h1>
    <h2>Just item</h2>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>with `disabled=false` on top-level</h2>
    <OkuAccordion type="single" class="accordion-root" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item" :disabled="item === 'Two'">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>Force mounted contents</h1>
    <OkuAccordion type="single" class="accordion-root">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item" :value="item">
        <OkuAccordionHeader class="accordion-header">
          <OkuAccordionTrigger class="accordion-trigger">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content" force-mount>
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h1>State attributes</h1>
    <h2>Accordion disabled</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" disabled>
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion enabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="false">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item)">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
    <h2>Accordion disabled with item override</h2>
    <OkuAccordion type="single" class="accordion-root-attr" default-value="Two" :disabled="true">
      <OkuAccordionItem v-for="item in values" :key="item" class="accordion-item-attr" :value="item" :disabled="['Two', 'Four'].includes(item) ? false : undefined">
        <OkuAccordionHeader class="accordion-header-attr">
          <OkuAccordionTrigger class="accordion-trigger-attr">
            {{ item }}
          </OkuAccordionTrigger>
        </OkuAccordionHeader>
        <OkuAccordionContent class="accordion-content-attr">
          {{ item }}: Per erat orci nostra luctus sociosqu mus risus penatibus, duis elit vulputate
          viverra integer ullamcorper congue curabitur sociis, nisi malesuada scelerisque
          quam suscipit habitant sed.
        </OkuAccordionContent>
      </OkuAccordionItem>
    </OkuAccordion>
  </div>
</template>
