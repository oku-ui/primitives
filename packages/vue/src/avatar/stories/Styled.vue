<script setup lang="ts">
import { OkuAvatar, OkuAvatarFallback, OkuAvatarImage } from '@oku-ui/avatar'
import AvatarIcon from './AvatarIcon.vue'

const src = 'https://picsum.photos/id/1005/400/400'
const srcBroken = 'https://broken.link.com/broken-pic.jpg'
</script>

<template>
  <h1>Without image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarFallback class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage class="avatar-image" alt="John Smith" :src="src" />
    <OkuAvatarFallback :delay-ms="300" class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback (but broken src)</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage
      class="avatar-image"
      alt="John Smith"
      :src="srcBroken"
      @loading-status-change="console.log($event)"
    />
    <OkuAvatarFallback class="avatar-fallback">
      <AvatarIcon />
    </OkuAvatarFallback>
  </OkuAvatar>
</template>
