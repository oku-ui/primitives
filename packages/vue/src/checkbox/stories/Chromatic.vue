<script setup lang="ts">
import { OkuCheckbox, OkuCheckboxIndicator } from '@oku-ui/checkbox'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" default-checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Controlled</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox" :checked="false">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox" checked>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Indeterminate</h1>
  <OkuCheckbox class="checkbox" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Disabled</h1>
  <OkuCheckbox class="checkbox" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Force mounted indicator</h1>
  <OkuCheckbox class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>

  <h1>State attributes</h1>
  <h2>Unchecked</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Checked</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Indeterminate</h2>
  <OkuCheckbox class="checkbox-attr-styles" checked="indeterminate">
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Disabled</h2>
  <OkuCheckbox class="checkbox-attr-styles" default-checked disabled>
    <OkuCheckboxIndicator class="checkbox-attr-styles" />
  </OkuCheckbox>

  <h2>Force mounted indicator</h2>
  <OkuCheckbox class="checkbox-attr-styles">
    <OkuCheckboxIndicator class="checkbox-attr-styles" force-mount :style="{ height: '20px' }" />
  </OkuCheckbox>
</template>
