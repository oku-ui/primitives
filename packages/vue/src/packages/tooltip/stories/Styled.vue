<script setup lang="ts">
import {
  OkuTooltip,
  OkuTooltipArrow,
  OkuTooltipContent,
  OkuTooltipPortal,
  OkuTooltipProvider,
  OkuTooltipTrigger,
} from '@oku-ui/tooltip'
</script>

<template>
  <OkuTooltipProvider>
    <OkuTooltip>
      <OkuTooltipTrigger class="tooltip-trigger">
        Hover or Focus me
      </OkuTooltipTrigger>
      <OkuTooltipPortal>
        <OkuTooltipContent class="tooltip-content" :side-offset="5">
          Nicely done!
          <OkuTooltipArrow class="tooltip-arrow" :offset="10" />
        </OkuTooltipContent>
      </OkuTooltipPortal>
    </OkuTooltip>
  </OkuTooltipProvider>
</template>
