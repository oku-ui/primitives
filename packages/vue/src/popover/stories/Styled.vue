<script setup lang="ts">
import {
  OkuPopover,
  OkuPopoverArrow,
  OkuPopoverClose,
  OkuPopoverContent,
  OkuPopoverPortal,
  OkuPopoverTrigger,
} from '@oku-ui/popover'
</script>

<template>
  <div
    :style="{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      height: '200vh',
    }"
  >
    <OkuPopover>
      <OkuPopoverTrigger class="popover-trigger">
        open
      </OkuPopoverTrigger>
      <OkuPopoverPortal>
        <OkuPopoverContent class="popover-content" :side-offset="5">
          <OkuPopoverClose class="popover-close">
            close
          </OkuPopoverClose>
          <OkuPopoverArrow class="popover-arrow" :width="20" :height="10" />
        </OkuPopoverContent>
      </OkuPopoverPortal>
    </OkuPopover>

    <input>
  </div>
</template>
