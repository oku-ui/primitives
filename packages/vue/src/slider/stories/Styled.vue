<script setup lang="ts">
import { OkuSlider, OkuSliderRange, OkuSliderThumb, OkuSliderTrack } from '@oku-ui/slider'
</script>

<template>
  <div>
    <OkuSlider class="slider-rootSlider">
      <OkuSliderTrack class="slider-track">
        <OkuSliderRange class="slider-range" />
      </OkuSliderTrack>
      <OkuSliderThumb class="slider-thumb" />
    </OkuSlider>
  </div>
</template>
