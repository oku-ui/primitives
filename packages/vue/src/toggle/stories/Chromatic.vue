<script setup lang="ts">
import { OkuToggle } from '@oku-ui/toggle'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Off</h2>
  <OkuToggle class="toggle">
    Toggle
  </OkuToggle>

  <h2>On</h2>
  <OkuToggle class="toggle" default-pressed>
    Toggle
  </OkuToggle>

  <h1>Controlled</h1>
  <h2>Off</h2>
  <OkuToggle class="toggle" :pressed="false">
    Toggle
  </OkuToggle>

  <h2>On</h2>
  <OkuToggle class="toggle" pressed>
    Toggle
  </OkuToggle>

  <h1>Disabled</h1>
  <OkuToggle class="toggle" disabled>
    Toggle
  </OkuToggle>

  <h1>State attributes</h1>
  <OkuToggle class="toggle-attr-styles">
    Toggle
  </OkuToggle>
  <OkuToggle class="toggle-attr-styles" disabled>
    Toggle
  </OkuToggle>
</template>
