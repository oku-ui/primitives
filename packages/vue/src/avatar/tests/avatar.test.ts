import { enableAutoUnmount, mount, shallowMount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuAvatar, OkuAvatarFallback, OkuAvatarImage } from '../'

import Styled from '../stories/Styled.vue'
import Chromatic from '../stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const AVATAR_TEST_ID = 'avatar-test'
const FALLBACK_TEXT = 'AB'
const IMAGE_ALT_TEXT = 'Fake Avatar'
const DELAY = 300

describe('okuAvatar', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount({
      components: {
        OkuAvatar,
        OkuAvatarFallback,
      },
      setup() {
        return {
          AVATAR_TEST_ID,
          FALLBACK_TEXT,
        }
      },
      template: `
        <OkuAvatar :data-testid="AVATAR_TEST_ID">
          <OkuAvatarFallback>{{ FALLBACK_TEXT }}</OkuAvatarFallback>
        </OkuAvatar>
      `,
    }, {
      attachTo: document.body,
    })
  })

  it('should render OkuAvatar correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuAvatar).html()).toMatchSnapshot()
  })

  it('should render OkuAvatarImage correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => shallowMount(OkuAvatarImage)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should render OkuAvatarFallback correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => shallowMount(OkuAvatarFallback)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuAvatar)" not found.')
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  describe('given an Avatar with fallback and a working image', () => {
    let wrapper: VueWrapper
    const originalGlobalImage = window.Image

    beforeAll(() => {
      (window.Image as any) = class MockImage {
        onload: () => void = () => {}
        src: string = ''
        constructor() {
          setTimeout(() => {
            this.onload()
          }, DELAY)
          return this
        }
      }
    })

    beforeEach(() => {
      wrapper = mount({
        components: {
          OkuAvatar,
          OkuAvatarFallback,
          OkuAvatarImage,
        },
        setup() {
          return {
            AVATAR_TEST_ID,
            FALLBACK_TEXT,
            IMAGE_ALT_TEXT,
          }
        },
        template: `
          <OkuAvatar :data-testid="AVATAR_TEST_ID">
            <OkuAvatarFallback>{{ FALLBACK_TEXT }}</OkuAvatarFallback>
            <OkuAvatarImage src="https://picsum.photos/id/1005/400/400" :alt="IMAGE_ALT_TEXT" />
          </OkuAvatar>
        `,
      }, {
        attachTo: document.body,
      })
    })

    afterAll(() => {
      window.Image = originalGlobalImage
    })

    it('should render the fallback initially', () => {
      expect(wrapper.text()).toContain(FALLBACK_TEXT)
    })

    it('should not render the image initially', () => {
      expect(wrapper.find('img').exists()).toBe(false)
    })

    it('should render the image after it has loaded', async () => {
      await new Promise(resolve => setTimeout(resolve, DELAY))

      expect(wrapper.find('img').exists()).toBe(true)
    })

    it('should have alt text on the image', async () => {
      await new Promise(resolve => setTimeout(resolve, DELAY))

      expect(wrapper.find('img').attributes('alt')).toBe(IMAGE_ALT_TEXT)
    })
  })

  describe('given an Avatar with fallback and delayed render', () => {
    let wrapper: VueWrapper

    beforeEach(() => {
      wrapper = mount({
        components: {
          OkuAvatar,
          OkuAvatarFallback,
        },
        setup() {
          return {
            AVATAR_TEST_ID,
            FALLBACK_TEXT,
            IMAGE_ALT_TEXT,
            DELAY,
          }
        },
        template: `
          <OkuAvatar :data-testid="AVATAR_TEST_ID">
            <OkuAvatarFallback :delayMs="DELAY">{{ FALLBACK_TEXT }}</OkuAvatarFallback>
          </OkuAvatar>
        `,
      }, {
        attachTo: document.body,
      })
    })

    it('should not render a fallback immediately', () => {
      expect(wrapper.text()).not.toContain(FALLBACK_TEXT)
    })

    it('should render a fallback after the delay', async () => {
      expect(wrapper.text()).not.toContain(FALLBACK_TEXT)

      await new Promise(resolve => setTimeout(resolve, DELAY))

      expect(wrapper.text()).toContain(FALLBACK_TEXT)
    })
  })
})

describe('okuAvatar Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
