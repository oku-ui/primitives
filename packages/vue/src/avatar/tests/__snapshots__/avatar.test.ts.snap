// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuAvatar > should render OkuAvatar correctly 1`] = `"<oku-avatar-stub is="span" aschild="false" data-testid="avatar-test"></oku-avatar-stub>"`;

exports[`okuAvatar > should render OkuAvatar correctly 2`] = `"<oku-primitive-stub is="span" aschild="false"></oku-primitive-stub>"`;

exports[`okuAvatar > should render OkuAvatarFallback correctly 1`] = `[Error: \`OkuAvatar\` must be used within \`OkuAvatar\`]`;

exports[`okuAvatar > should render OkuAvatarImage correctly 1`] = `[Error: \`OkuAvatar\` must be used within \`OkuAvatar\`]`;

exports[`okuAvatar > should render OkuAvatarImage correctly 2`] = `"[Vue warn]: Missing required prop: "src""`;

exports[`okuAvatar Stories > chromatic > should render correctly 1`] = `
"<h1>Without image &amp; with fallback</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>
<h1>With image &amp; with fallback</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>
<h1>With image &amp; with fallback (but broken src)</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>"
`;

exports[`okuAvatar Stories > styled > should render correctly 1`] = `
"<h1>Without image &amp; with fallback</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>
<h1>With image &amp; with fallback</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>
<h1>With image &amp; with fallback (but broken src)</h1>
<oku-avatar-stub is="span" aschild="false" class="avatar"></oku-avatar-stub>"
`;
